### ISSUE TITLE
1️⃣ Part A.1: Add Best Model Saving Support

### ISSUE THREAD

**Issue by kjpou1** [kjpou1/regimetry#10]
#### 🎯 Goal:

Enable saving the best-performing model (by validation loss or another metric) via a `ModelCheckpoint` callback — alongside the standard final model.

#### 📦 Deliverables:

* [ ] Add `save_best_model: true` and `best_model_filename: best_model.keras` to `TrainingProfileConfig`
* [ ] Register a `ModelCheckpoint` callback in `ForecastModelBuilderService` if `save_best_model` is true
* [ ] Ensure path joins to `output_dir` if a relative name is provided
* [ ] Log both final and best model paths after training completes

---

thoughts maybe something like this:

  model_checkpoint:
    enabled: true
    monitor: val_loss
    mode: min
    save_best_only: true
    save_weights_only: false
    filename: best_model.keras

---

### 🧠 Notes:

* This should *not* overwrite the final model unless explicitly requested.
* Defaults should be conservative: always keep `forecast_model.keras`, and optionally add `best_model.keras`.
* Consider supporting `monitor` and `mode` fields (`val_loss`, `min`) in training profile if needed later.
